tmt/convert.py

```diff
@@ -653,12 +700,13 @@ def read_nitrate_case(testcase, makefile_data=None, general=False):
                 testcase.tester.name, testcase.tester.email)
         else:
             try:
+                assert makefile_data is not None
                 # Use contact from Makefile if it's there and email matches
                 if re.search(testcase.tester.email, makefile_data['contact']):
                     data['contact'] = makefile_data['contact']
                 else:
                     raise KeyError
-            except (KeyError, TypeError):
+            except (KeyError, TypeError, AssertionError):
```

Comment:
Nope, never. `assert` is not executed when running Python with `-O` option, which means this exception would never be raised. `assert` helps mypy understand what's happening, but don't rely on it when it comes to the function of the code. You may use different forms of checks if `makefile_data == None` needs to be sanitized somehow.